### ISSUE TITLE
Custom .env file path option for build time in docker compose config

### ISSUE THREAD

**Issue by twistezo** [docker/compose#9107]
**Feature request**

tl;dr 
`env_file` option in docker compose build-time context like in run-time context.

**Description**

Project structure:
```
foo/
   Dockerfile
   .env
docker-compose.yml
.env
```

If we want to use custom path for `.env` file in container's:
- run-time 

  We have option `env_file` for `docker-compose.yml` file which can be set ex. `env_file: foo/.env` and all env variables from this file are visible inside container.
- build-time

  We can't do this. We have access only to `.env` file which should be next to `docker-compose.yml` and we have to do extra effort for combine env variables like below: 

```
# .env (next to docker-compose.yml)
TEST=123
```

```
# docker-compose.yml
...
   build:
      args: 
         - TEST=$TEST
```

```
# Dockerfile
ARG TEST
```

---

It will be much simpler if we had `env_file` option for `build` context in docker compose config without extra effort.
```
# docker-compose.yml
...
   build:
      env_file: foo/.env
```

**Comment by ndeloof** [comment]
That's a nice suggestion, should be discussed as part of the [compose specification](https://github.com/compose-spec/compose-spec)

**Comment by 2kCarlos** [comment]
This would be awesome for my use case as well!
Any update(s) on this?

**Comment by Mifrill** [comment]
📝 workaround:

Project structure:
```
foo/
   Dockerfile
docker-compose.yml
foo.build-args
launch.sh
```
`# .args (next to docker-compose.yml)`
```
# my test comment
TEST=123
```
`# docker-compose.yml`
```yml
   build:
      args: 
         - TEST=$TEST
```
`# Dockerfile`
```dockerfile
ARG TEST=${TEST:-bla}
```
`# docker-compose.yml`
```yml
   build:
      env_file: foo/.env
```
create new file `# launch.sh`
```bash
#!/usr/bin/env bash

source ./.env

shopt -s nullglob
for file in *.build-args; do
  while IFS= read -r line; do
    if [[ $line == \#* ]]; then
      continue
    fi
    export "$line"
  done < "$file"
done

docker-compose up
```

Use:
`./launch.sh` instead `docker-compose up` to run.

**Comment by b-a0** [comment]
Just ran into this issue as well, would be great if there would be feature parity between the `--env-file` CLI option and the `env_file` yaml key.
